rewrite-maven/src/main/java/org/openrewrite/maven/tree/Profile.java

```diff
@@ -44,7 +44,18 @@ public class Profile {
     List<Plugin> plugins;
     List<Plugin> pluginManagement;
 
-    public boolean isActive(Iterable<String> activeProfiles) {
-        return ProfileActivation.isActive(id, activeProfiles, activation);
```

Comment:
In general we're fairly conservative with changes to any `tree.*` classes, as these are also serialized, with a more high impact change for our users. Is there anything we can do to restore the old methods to minimize the impact?

Reply:
Sorry @timtebeek I didn't see this.  Let me check!

Reply:
We're talking about standard object serialization, right?  I didn't think methods like the ones I removed/added played a role in that.  I thought it was more about state.

Regardless, let me see what I can easily put back.

Reply:
It's a little more involved, but ideally we minimize changes to the API where we can. Thanks for having a look!

Reply:
Makes sense.  I should have thought about public API.  I pushed some changes.  Let me know what you think.

Reply:
little :bump: on this:)

Reply:
Thanks! I'll need to take another look; In general I can see we've recently started a trend of moving higher-order concepts to Traits classes, away from the "dumb" lossless semantic tree elements that serve mostly to faithfully reproduce the underlying source code. I just haven't been able to allocate time to see if that pattern could/should be applied here too, as opposed to the `implements WithProfiles.Profile` we have now. Still strive to get this in, but balancing a lot of work. :)

Reply:
If there's anything I can do to help, just let me know!  I don't mind copying any existing pattern you have accepted.

Reply:
Any ideas on how we can help with this?  We have some workarounds, but they don't work quite as well as we thought.

Reply:
:bump :)

Reply:
Apologies for the delayed response; it's hard to get to everything these days. I think to start revert the `implements WithProfiles` on the `..tree.*` classes here would help get this in a state that's closer to review & merge. Not sure about what the exact replacement should be, but that would help restart this, along with resolving the conflicts.

Reply:
I understand.  I will see if I can get rid of the interface.  I know I can if I just copy code around.  I looked at `Trait`, but if there's an obvious application here, I'm not seeing it.  Seems oriented around a cursor, and that doesn't seem to be playing a role.

Anyhow, let me see what I can come up with over the next few days.

Reply:
Ok, I pushed a change that does not involve profiles.  Unfortunately, I think Maven settings will still have an issue (the interface-approach tried to deal with that).  However, I don't _think_ that Maven settings problem affects us.  I'll have to verify.